zerver/tests/test_user_groups.py

```diff
@@ -352,39 +353,82 @@ def test_update_members_of_user_group(self) -> None:
+        )
+        self.assertEqual(msg_event[0]["event"]["message_dict"]["content"], expected_notification)
+
         # Test adding a member already there.
         result = self.client_post(f"/json/user_groups/{user_group.id}/members", info=params)
         self.assert_json_error(result, f"User {othello.id} is already a member of this group")
         self.assertEqual(UserGroupMembership.objects.filter(user_group=user_group).count(), 2)
         members = get_direct_memberships_of_users(user_group, [hamlet, othello])
         self.assert_length(members, 2)
 
+        # Test user adding itself to user group
+        desdemona = self.example_user("desdemona")
+        self.login_user(desdemona)
+        add = [desdemona.id]
+        events = []
+
+        params = {"add": orjson.dumps(add).decode()}
+        with self.tornado_redirected_to_list(events, expected_num_events=1):
+            result = self.client_post(f"/json/user_groups/{user_group.id}/members", info=params)
+            self.assert_json_success(result)
+
+        self.assertEqual(UserGroupMembership.objects.filter(user_group=user_group).count(), 3)
+        members = get_direct_memberships_of_users(user_group, [hamlet, othello, desdemona])
+        self.assert_length(members, 3)
+
+        # No notification message is sent for adding to user group
+        msg_event = [e for e in events if e["event"]["type"] == "message"]
+        self.assert_length(msg_event, 0)
+
         aaron = self.example_user("aaron")
 
         # For normal testing we again log in with hamlet
         self.logout()
         self.login_user(hamlet)
         # Test remove members
+        events = []
+
         params = {"delete": orjson.dumps([othello.id]).decode()}
-        result = self.client_post(f"/json/user_groups/{user_group.id}/members", info=params)
-        self.assert_json_success(result)
-        self.assertEqual(UserGroupMembership.objects.filter(user_group=user_group).count(), 1)
-        members = get_direct_memberships_of_users(user_group, [hamlet, othello, aaron])
-        self.assert_length(members, 1)
+        with self.tornado_redirected_to_list(events, expected_num_events=2):
+            result = self.client_post(f"/json/user_groups/{user_group.id}/members", info=params)
+            self.assert_json_success(result)
+        self.assertEqual(UserGroupMembership.objects.filter(user_group=user_group).count(), 2)
+        members = get_direct_memberships_of_users(user_group, [hamlet, othello, aaron, desdemona])
+        self.assert_length(members, 2)
+
+        # A notification message is sent for removing from user group
+        msg_event = [e for e in events if e["event"]["type"] == "message"]
+        self.assert_length(msg_event, 1)
+        expected_notification = (
+            f"{silent_mention_syntax_for_user(hamlet)} unsubscribed you from the group @_*support*."
+        )
+        self.assertEqual(msg_event[0]["event"]["message_dict"]["content"], expected_notification)
```

Comment:
Here also would be good to test the case when removing yourself from user group.